test/unit-tests/broker-launch.test.js

```diff
@@ -1,26 +1,66 @@
+const sinon = require('sinon');
+const cp = require('child_process');
+const { EventEmitter } = require('events');
 const expect = require('chai').expect;
-const io = require('socket.io-client');
 
-describe('Broker connection handling', () => {
-    let client, dashboardServer;
+const launchBroker = require('../../dashboard/index');
 
-    before((done) => {
-        dashboardServer = require('../../dashboard/server/index');
-        client = io('http://localhost:5001/');
-        client.on('connect', done);
+// mock event emitter to simulate ChildProcess object
+const mockChildProcess = new EventEmitter();
+
+// sinon spies for behaviour verification
+let unrefSpy = sinon.spy();
+let disconnectSpy = sinon.spy();
+
+describe('Broker daemonized launching', () => {
```

Comment:
Nitpick: Ideally this should be a noun for the unit that we're testing.
```suggestion
describe('Broker', () => {
```

If this cannot be a noun, there must be something wrong with how we've written module that we're testing. 🤔

Reply:
Actually I was just bad with unit test names. I read a few articles about naming conventions and I think ```Broker daemon``` is the name for this as that is the module it tests!